core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class ViewMetadata implements Serializable {
+  static final int DEFAULT_VIEW_FORMAT_VERSION = 1;
+  static final int SUPPORTED_VIEW_FORMAT_VERSION = 1;
+
+  public static ImmutableViewMetadata.Builder buildFrom(ViewMetadata metadata) {
```

Comment:
is this used anywhere? Also we can probably just use `ImmutableViewMetadata.builder().from(metadata).xyz().build()` here

Reply:
or is the goal here to just create a copy? In that case we can also just do `ImmutableViewMetadata.copyOf(metadata)`

Reply:
I guess this is trying to follow `TableMetadata.buildFrom()`?

Reply:
@stevenzwu `ImmutableViewMetadata` is a generated class from the `@Value.Immutable` annotation. See also https://immutables.github.io/ for some additional details